nbclient/client.py

```diff
@@ -245,6 +256,52 @@ class NotebookClient(LoggingConfigurable):
 
     kernel_manager_class: KernelManager = Type(config=True, help='The kernel manager class to use.')
 
+    on_execution_start: t.Optional[t.Callable] = Callable(
```

Comment:
I think `on_notebook_start` might be more intuitive. `on_execution_start` makes me think that it will be called as soon as the execution begins, whereas this is actually called when the notebook is initialized but no cells have been identified for execution yet.